docgenerator/src/main/scala/codacy/codenarc/docs/DocGenerator.scala

```diff
@@ -0,0 +1,311 @@
+package codacy.codenarc.docs
+
+import java.lang.reflect.Field
+
+import better.files.File
+import com.codacy.plugins.api.results.{Parameter, Pattern, Result, Tool}
+import play.api.libs.json.Json
+
+import scala.util.matching.Regex
+import scala.jdk.CollectionConverters._
+import org.reflections.Reflections
+
+import scala.annotation.tailrec
+import scala.util.Try
+
+object DocGenerator {
+  private val toolName = "codenarc"
+
+  private val codeNarcVersionFile = ".codenarc-version"
+  private val codeNarcGitRepository = "github.com/CodeNarc/CodeNarc"
+  private val patternsJsonFile = "patterns.json"
+  private val descriptionJsonFile = "description.json"
+  private val docsFolder = "docs"
+  private val descriptionFolder = "description"
+
+  private val codeNarcDocumentationStartingPoint = "codenarc-rule-index.md"
+
+  private val reflections = new Reflections("org.codenarc.rule")
+  private val subTypes = reflections.getSubTypesOf(classOf[org.codenarc.rule.AbstractAstVisitorRule]).asScala
+
+  case class RuleInformation(
+      patternId: Pattern.Id,
+      description: String,
+      descriptionExtended: String = "",
+      priority: Option[RulePriority],
+      parameters: Option[Set[String]]
+  )
+
+  case class RulePriority(priority: Option[String], priorityType: Option[String])
+
+  private val defaultPriority = "3"
+  private val defaultRuleType = "CodeStyle"
+  private val defaultRulePriority = RulePriority(Some(defaultPriority), Some(defaultRuleType))
+
+  /**
+    * Converts from CodeNarc's complexity to Codacy level
+    */
+  val levelFromPriority: PartialFunction[Option[String], Result.Level.Value] = {
+    case Some("1") => Result.Level.Err
+    case Some("2") => Result.Level.Warn
+    case Some("3") => Result.Level.Info
+    case _ => Result.Level.Info
+  }
+
+  val categoryTypeFromCategory: PartialFunction[Option[String], Pattern.Category.Value] = {
+    case Some("security") => Pattern.Category.Security
+    case Some("unused") => Pattern.Category.UnusedCode
+    case _ => Pattern.Category.CodeStyle
+  }
+
+  /**
+    * Convert from list of parameters to Parameter.Specification list
+    */
+  def parameterSpecificationFromParametersStringList(
+      parameters: Option[Set[String]]
+  ): Option[Set[Parameter.Specification]] =
+    parameters.map(
+      paramList => paramList.map(param => Parameter.Specification(Parameter.Name(param), Parameter.Value("")))
+    )
+
+  def parameterDescriptionFromParametersStringList(
+      parameters: Option[Set[String]]
+  ): Option[Set[Parameter.Description]] =
+    parameters.map(
+      paramList => paramList.map(param => Parameter.Description(Parameter.Name(param), Parameter.DescriptionText("")))
+    )
+
+  /**
+    * returns the regex to get information from markdown file for a rule
+    *
+    * @param ruleName Name of the rule to get information
+    * @return
+    */
+  def ruleInfoRegex(ruleName: String) = new Regex("## " + ruleName + "([\\n\\S\\s\\w\\d][^##])*")
```

Comment:
You can create `Regex`es calling `.r` on `String`s:
```scala
"hello".r
```

Reply:
Done ✅